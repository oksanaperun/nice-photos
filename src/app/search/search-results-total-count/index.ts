export * from './search-results-total-count.component';
