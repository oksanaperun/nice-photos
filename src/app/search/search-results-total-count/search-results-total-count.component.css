h5 {
  font-family: 'Nunito', serif;
  color: rgba(0, 0, 0, 0.5);
  text-align: center;
  margin-top: 5px;
  margin-bottom: 10px;
}
