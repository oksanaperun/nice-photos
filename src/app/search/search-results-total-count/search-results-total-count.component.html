<h5 i18n>{totalCount, plural, =0 {No photos} =1 {1 photo} other {{{totalCount}} photos}} found</h5>
