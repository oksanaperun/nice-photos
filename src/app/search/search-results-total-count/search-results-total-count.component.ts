import { Component, Input, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-search-results-total-count',
  templateUrl: './search-results-total-count.component.html',
  styleUrls: ['./search-results-total-count.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})

export class SearchResultsTotalCountComponent {
  @Input() totalCount: number;
}
