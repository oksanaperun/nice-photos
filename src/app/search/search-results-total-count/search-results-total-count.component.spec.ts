import { TestBed, ComponentFixture } from '@angular/core/testing';
import { SearchResultsTotalCountComponent } from './search-results-total-count.component';

describe('SearchResultsTotalCountComponent', () => {
  let component: SearchResultsTotalCountComponent;
  let fixture: ComponentFixture<SearchResultsTotalCountComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [SearchResultsTotalCountComponent]
    });

    fixture = TestBed.createComponent(SearchResultsTotalCountComponent);
    component = fixture.componentInstance;
  });

  it('should be rendered correctly when total items count equals 1', () => {
    component.totalCount = 1;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should be rendered correctly when total items count is greater than 1', () => {
    component.totalCount = 10;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});