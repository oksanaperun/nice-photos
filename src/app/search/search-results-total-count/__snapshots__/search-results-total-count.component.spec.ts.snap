// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsTotalCountComponent should be rendered correctly when total items count equals 1 1`] = `
<app-search-results-total-count
  totalCount={[Function Number]}
>
  <h5>
    
    
    
    1 photo
    
     found
  </h5>
  
</app-search-results-total-count>
`;

exports[`SearchResultsTotalCountComponent should be rendered correctly when total items count equals zero 1`] = `
<app-search-results-total-count
  totalCount="0"
>
  <h5>
    
    
    No photos
    
    
     found
  </h5>
  
</app-search-results-total-count>
`;

exports[`SearchResultsTotalCountComponent should be rendered correctly when total items count is greater than 1 1`] = `
<app-search-results-total-count
  totalCount={[Function Number]}
>
  <h5>
    
    
    
    
    10 photos
     found
  </h5>
  
</app-search-results-total-count>
`;
